primer.md

```diff
@@ -238,23 +238,69 @@ there are errors during the processing of a CloudEvent, the software
-particular version of a schema document. Often these attribute values will then
-distinguish each variant by including some version-specific string as part of
-its value. For example, a version number (`v1`, `v2`), or a date (`2018-01-01`)
-might be used.
-
-The CloudEvents specification does not mandate any particular pattern to be
-used, or even the use of version strings at all. This decision is up to each
-event producer. However, when a version-specific string is included, care should
-be taken whenever its value changes as event consumers might be reliant on the
-existing value and thus a change could be interpreted as a "breaking change".
-Some form of communication between producers and consumers should be established
-to ensure the event consumers know what possible values might be used. In
-general, this is true for all CloudEvents attributes as well.
+## Versioning of CloudEvents
+
+For many CloudEvents, the schema of the data within the event might
+change over time. The CloudEvents specification does not mandate any
+particular pattern to be used, or even the use or consideration of
+versioning at all. This decision is up to each event producer.
+
+However, event producers are encouraged to consider how they might
+evolve their schema without breaking consumers. Two specific
+context attributes are particularly important in this respect, but
+differ in expected usage: `type` and `dataschema`. The differences
+between these attributes with respect to versioning are described
+below.
+
+Event producers are encouraged to consider versioning from the
+outset, before first declaring a particular CloudEvent to be
+"stable". Documentation of the chosen versioning scheme, including
+the rationale behind it, is likely to be appreciated by consumers.
+
+### The role of the `type` attribute within versioning
+
+The `type` attribute is expected to be the primary means by which
+consumers identify the type of event that they receive. This could
+be achieved by subscribing to a specific CloudEvent type, or by
+filtering all received CloudEvents by type locally. But consumers
+who have identified a CloudEvent type will generally expect the data
+within that type to only change in backwardly-compatible ways,
+unless clearly indicated otherwise. The precise meaning of
+"backwardly-compatible" will vary by data content type.
+
+When a CloudEvent's data changes in a backwardly-compatible way, the
+value of the `type` attribute should generally stay the same.
+
+When a CloudEvent's data changes in a backwardly-incompatible way,
+the value of the `type` attribute should generally change. The event
+producer is encouraged to produce both the old event and the new
+event for some time (potentially forever) in order to avoid
+disrupting consumers.
+
+When considering the value of the `type` attribute, event producers
+may choose any versioning scheme they wish, such as a version number
```

Comment:
overall this seems ok, but it does drop the idea of no versioning scheme at all for "type". Was that intentional? The use of the phrase "may choose any" kind of implies they need to choose one. The old text implied it was optional.

Reply:
No, that wasn't intentional - although I do want to encourage it.

How would this be:

> When considering the value of the `type` attribute, event producers
> may choose any versioning scheme they wish, such as a version number
> (`v1`, `v2`) or a date (`2018-01-01`) as part of the value. Alternatively, they
> may choose not to include a version in the `type` value at all.

Reply:
sure that works - thanks!

Reply:
Done in an extra commit. Thanks for the careful review :)